openmc/source.py

```diff
@@ -124,16 +230,25 @@ def __init__(
             self.domain_ids = [d.id for d in domains]
 
     @property
-    def file(self):
-        return self._file
-
-    @property
-    def library(self):
-        return self._library
+    def type(self) -> str:
+        return 'independent'
+
+    def __getattr__(self, name):
+        cls_names = {'file': 'FileSource', 'library': 'CompiledSource',
+                    'parameters': 'CompiledSource'}
+        if name in cls_names:
+            raise AttributeError(
+                f'The "{name}" attribute has been deprecated on the '
+                f'IndependentSource class. Please use the {cls_names[name]} class.')
+        else:
+            super().__getattr__(name, value)
```

Comment:
This doesn't appear to work:
```python
>>> import openmc
>>> src = openmc.IndependentSource()
>>> src.frisbee
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/romano/openmc/openmc/source.py", line 244, in __getattr__
    super().__getattr__(name, value)
AttributeError: 'super' object has no attribute '__getattr__'. Did you mean: '__setattr__'?
```

Reply:
Ah yeah there were a couple of problems there. Taken care of now and I've added another test check that accessing attributes that don't exist fails correctly.